src/appleseed.studio/mainwindow/mainwindow.cpp

```diff
@@ -2138,6 +2151,24 @@ void MainWindow::slot_fullscreen()
         button->set_fullscreen(m_fullscreen);
 }
 
+void MainWindow::slot_check_fullscreen()
+{
+    QAction* fullscreen = m_ui->menu_view->findChild<QAction*>("Fullscreen");
+
+    if (all_of(m_minimize_buttons.cbegin(),
+               m_minimize_buttons.cend(),
+               [](MinimizeButton* button) {return button->is_on();})) {
```

Comment:
In our coding conventions, all braces go on their own lines.

Reply:
Is the header file declaring `all_of()` included?

Reply:
The logic seems convoluted, what about something like

    const bool is_fullscreen = all_of(...);
    fullscreen->setChecked(is_fullscreen);

Also, are you sure all buttons must be `on`? Shouldn't it be off?

Reply:
all_of is coming from using namespace std. 

The buttons work opposite than what's expected. On returns 0 and off returns 1. This can be changed but I think there's more potential for new bugs. Do you want me to dig into this?

Reply:
> all_of is coming from using namespace std.

Yes but that's not the question I was asking :) `std::all_of()` is declared in `<algorithm>`. It so happens that it's already included so we're good here.

Reply:
Got it, thanks!